### PR TITLE
chore: add no-unsafe-optional-chaining as it became part of recommanded settings

### PR DESCRIPTION
**Pull request by peter-sanderson**
https://eslint.org/docs/latest/rules/no-unsafe-optional-chaining became part of recommended settings

Prep, for https://github.com/trezor/trezor-suite/pull/15144

### CODE REVIEW

File: packages/components/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export const Dropdown = forwardRef(
                 return;
             }
 
-            const { width, height } = menuRef.current?.getBoundingClientRect();
```
Comment (peter-sanderson) [inline packages/components/src/components/Dropdown/Dropdown.tsx:127]: Its checked above, `!menuRef.current` => early return, the `?` not needed

File: packages/components/src/components/typography/TruncateWithTooltip/TruncateWithTooltip.tsx [hunk 2]
```diff
@@ -29,8 +29,13 @@ export const TruncateWithTooltip = ({
     useEffect(() => {
         if (!containerRef.current || !scrollWidth || !scrollHeight) return;
         const resizeObserver = new ResizeObserver(entries => {
-            const { inlineSize: elementWidth, blockSize: elementHeight } =
-                entries[0].borderBoxSize?.[0];
+            const borderBoxSize = entries[0].borderBoxSize?.[0];
+            if (!borderBoxSize) {
```
Comment (peter-sanderson) [inline packages/components/src/components/typography/TruncateWithTooltip/TruncateWithTooltip.tsx:33]: This probably never happens, but I wasn't sure

Comment (Lemonexe) [reply packages/components/src/components/typography/TruncateWithTooltip/TruncateWithTooltip.tsx:33]: Can't tell from documentation :/ But in any case, it's not worse than previous code, which throws error if `!borderBoxSize`

**Review by Lemonexe — APPROVED** [review]
LGTM

**Pull request merged**